src/scripts/genLttngProvider.py

```diff
@@ -546,7 +546,10 @@ def generateLttngFiles(etwmanifest,eventprovider_directory, dryRun):
                 lttngimpl_file.write("#include \"" + lttngevntheadershortname + "\"\n\n")
 
                 lttngimpl_file.write("""#ifndef tracepoint_enabled
-#define tracepoint_enabled(provider, name) FALSE
+
+extern "C" bool XplatEventLoggerIsEnabled();
```

Comment:
This means that if you turn on `COMPlus_EnableEventLog`, _all_ LTTng providers will report that they are turned on.  Can we use `XplatEventLogger::IsProviderEnabled(DOTNET_TRACE_CONTEXT providerCtx)` somehow to scope this down to what is actually enabled?  As it is, we are clobbering the configuration that can be set via `COMPlus_LTTngConfig` and saying that everything is on.  If we can't, I think this is sufficient, since it is still "pay to play".

Reply:
> This means that if you turn on COMPlus_EnableEventLog, all LTTng providers will report that they are turned on.

Not quite. Within the runtime the check is actually done via `ETW_EVENT_ENABLED` which does use the context structs you mentioned. The check via `tracepoint_enabled` happens after that within the event sink.

Reply:
Then this seems fine to me